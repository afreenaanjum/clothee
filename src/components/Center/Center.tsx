import React from 'react';
import {StyleSheet, View} from 'react-native';
import {IContainerProps} from '../../interfaces/interface';
const Center: React.FC<IContainerProps> = ({
  children,
  style,
}: IContainerProps) => {
  return <View style={[styles.center, style]}>{children}</View>;
};

const styles = StyleSheet.create({
  center: {
    flex: 1,
    alignSelf: 'stretch',
    justifyContent: 'center',
    alignItems: 'center',
  },
});

export default Center;
