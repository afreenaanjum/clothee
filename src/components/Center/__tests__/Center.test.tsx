import 'react-native';
import {Text} from 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Center from '../Center';

describe('Center widget', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Center children={<Text>Center</Text>} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when no children', () => {
    const tree = renderer.create(<Center children={undefined} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
