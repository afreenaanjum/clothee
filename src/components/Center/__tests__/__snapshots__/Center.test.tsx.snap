// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Center widget renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "flex": 1,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text>
    Center
  </Text>
</View>
`;

exports[`Center widget renders correctly when no children 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "flex": 1,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
/>
`;
